tests/scenario/conftest.py

```diff
@@ -0,0 +1,11 @@
+from unittest.mock import patch
+
+import pytest
+from charm import TraefikIngressCharm
+
+
+@pytest.fixture
+def traefik_charm():
+    with patch("charm.KubernetesServicePatch"):
+        with patch("lightkube.core.client.GenericSyncClient"):
+            yield TraefikIngressCharm
```

Comment:
TIL `yield` fixtures are recommended over `return` fixture because they let you add cleanup code.
So should we:
```python
svc_patcher = patch("charm.KubernetesServicePatch")
svc_patch = svc_patcher.start()

client_patcher = patch("lightkube.core.client.GenericSyncClient")
client_patch = client_patcher.start()

yield TraefikIngressCharm

svc_patcher.stop()
client_patcher.stop()
```

Reply:
AFAIAO this code is equivalent to the with ... yield statement I have. I believe the with ... context calls start() on enter and stop() on exit. No need to manually cleanup when you have contexts. Which is why I love contexts.